src/transformStateWithClones.js

```diff
@@ -65,7 +65,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const resulArr = [];
+  const copy = { ...state };
+  let add;
+  let remove;
+  let deleteAll;
+
+  for (const { operation, properties } of transforms) {
+    switch (operation) {
+      case 'addProperties':
+        for (const key in properties) {
+          copy[key] = properties[key];
+        }
+        add = { ...copy };
+        resulArr.push(add);
+        break;
+
+      case 'removeProperties':
+        for (const key of properties) {
+          delete copy[key];
+        }
+        remove = { ...copy };
+        resulArr.push(remove);
+        break;
+
+      case 'clear':
+        for (const key in copy) {
+          delete copy[key];
+        }
+        deleteAll = { ...copy };
+        resulArr.push(deleteAll);
+        break;
+    }
```

Comment:
You can add resulArr.push({ ...copy }) after your switch case, and then you won't need to duplicate it in every case.